src/mccabe/settings.rs

```diff
@@ -0,0 +1,28 @@
+//! Settings for the `mccabe` plugin.
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub struct Options {
+    pub max_complexity: Option<isize>,
+}
+
+#[derive(Debug, Hash)]
+pub struct Settings {
+    pub max_complexity: isize,
+}
+
+impl Settings {
+    pub fn from_options(options: Options) -> Self {
+        Self {
+            max_complexity: options.max_complexity.unwrap_or_default(),
+        }
+    }
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Self { max_complexity: -1 }
```

Comment:
I think we should default this to 10 or whatever Flake8's default is, and then use the check enablement to drive toggling on and off (rather than relying on both check enablement _and_ complexity value).

Reply:
flake8' has a similar behavior. Just selecting `C901` doesn't actually trigger the check, `max-complexity` still needs to be set to greater than `-1`:

```console
$ flake8 resources/test/fixtures/C901.py --max-complexity=2 --ignore=F
resources/test/fixtures/C901.py:15:1: C901 'if_elif_else_dead_path' is too complex (3)
resources/test/fixtures/C901.py:24:1: C901 'nested_ifs' is too complex (3)
resources/test/fixtures/C901.py:53:1: C901 'nested_functions' is too complex (3)
resources/test/fixtures/C901.py:61:1: C901 'try_else' is too complex (4)
resources/test/fixtures/C901.py:72:1: C901 'nested_try_finally' is too complex (3)
resources/test/fixtures/C901.py:82:1: C901 'foobar' is too complex (3)

$ flake8 resources/test/fixtures/C901.py --select=C901
```

but I'm all for whatever makes the most sense.

Reply:
Hmm yeah, I generally want to be compatible with existing tools but I do feel ok deviating on configuration settings. I think it's confusing that `--select=C901` doesn't _actually_ enable the check. I'd vote to use some kind of default and drive enablement purely through `select`.